tests/components/tasmota/test_discovery.py

```diff
@@ -45,61 +42,72 @@ async def test_subscribing_config_topic(hass, mqtt_mock):
     assert call_args[2] == 0
 
 
+async def test_valid_discovery_message(hass, mqtt_mock, caplog):
+    """Test discovery callback called."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+
+    with patch(
+        "homeassistant.components.tasmota.discovery.async_device_discovered"
```

Comment:
Don't patch the integration function. Patch the library.

Reply:
Fixed now